### PR TITLE
fix: bug in componentDidUpdate in base.tsx causing unnecessary rerenders

### PR DESCRIPTION
**Pull request by jaym910**
Hi great job writing this wrapper lib, its been super useful for me and my company. I fixed a bug that was causing an annoying visual bug where the plots would re-render when a parent component did, even when none of plot component's props were updated.

### CODE REVIEW

File: src/plots/base.tsx [hunk 1]
```diff
@@ -50,15 +65,25 @@ export default class BaseChart<
   componentDidUpdate() {
     const config = this.getConfig(this.props)
     const { data, ...restConfig } = config as any
-    const isConfigChanged = !isEqual(this.config, restConfig)
+    console.log(data)
```
Comment (kagawagao) [inline src/plots/base.tsx]: remove console

File: src/plots/base.tsx [hunk 2]
```diff
@@ -50,15 +65,25 @@ export default class BaseChart<
...
+    console.log(this.config.data)
+    const thisFnKeys = this.getFnKeys(this.config)
+    const restFnKeys = this.getFnKeys(restConfig)
+    const isConfigChanged = !isEqual(
+      omit(this.config, thisFnKeys),
+      omit(restConfig, restFnKeys)
+    )
+    const dataChanged = !isEqual(data, this.data)
```
Comment (kagawagao) [inline src/plots/base.tsx]: I don't think it is necessary to compare `data`

when the amount of data is large, this comparison can take a lot of time

Comment (jaym910) [reply src/plots/base.tsx]: Ok sure.

**Comment by kagawagao** [comment]
@jaym910 I'd have to revert this commit, thanks your contribution

if we filter function properties in config, chart will not be updated when events changed.

you can reproduce this bug in latet test case in `line.spec.tsx`

**Comment by jaym910** [comment]
Ok, sorry I didn't think of that scenario. And thanks for the tip

**Pull request merged**